### PR TITLE
[MI-2439] Enhancements and bug fixes

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
Enhancements:
1. Update the UI of Boards subscription notifications
2. Change the position of filter popover
3. Change the position of error message on subscription modal
4. Decrease the margin on "Add new subscription" button in RHS
5. Change the link color(Project name on subscription card) in dark theme on MM version 5.37.9

Bugs:
1. Infinite scroll not working in RHS

### CODE REVIEW

File: server/constants/constants.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ const (
 	// Regex to verify pull request link
 	PullRequestLinkRegex = `http(s)?:\/\/dev.azure.com\/[a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]*\/[a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]*\/_git\/[a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]*\/pullrequest\/[1-9]+`
 
+	WorkitemCommentedOnMarkdownRegex = ` commented on by [a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"|,.<>\/? ]*`
```
Comment (ayusht2810) [inline server/constants/constants.go:43]:
```suggestion
	WorkItemCommentedOnMarkdownRegex = ` commented on by [a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"|,.<>\/? ]*`
```

File: server/plugin/api.go [hunk 2]
```diff
@@ -510,12 +511,71 @@ func (p *Plugin) handleSubscriptionNotifications(w http.ResponseWriter, r *http.
...
+					Title: "Workitem Type",
+					Value: body.Resource.Fields.WorkItemType,
+				},
+			},
+			Footer:     body.Resource.Fields.ProjectName,
+			FooterIcon: fmt.Sprintf(constants.StaticFiles, p.GetSiteURL(), constants.PluginID, constants.FileNameProjectIcon),
+		}
+	case constants.SubscriptionEventWorkItemCommented:
+		reg := regexp.MustCompile(constants.WorkitemCommentedOnMarkdownRegex)
+		split := reg.Split(body.DetailedMessage.Markdown, -1)
```
Comment (ayusht2810) [inline server/plugin/api.go:542]: give some other name like message or text

File: webapp/src/components/modal/subComponents/modalSubtitleAndError/styles.scss [hunk 3]
```diff
@@ -10,3 +10,11 @@
     text-align: center;
   }
 }
+
+.modal__subtitle{
+  margin: 12px 0 0 0;
+}
+
+.modal__error{
+  margin: 12px 0 0 0;
```
Comment (ayusht2810) [inline webapp/src/components/modal/subComponents/modalSubtitleAndError/styles.scss:19]: can't we simply use `margin-top` if the rest of the values are 0 or are we overriding other values?

Comment (raghavaggarwal2308) [reply webapp/src/components/modal/subComponents/modalSubtitleAndError/styles.scss:19]: Yes, we are overriding the margin-bottom here

File: server/plugin/command.go [hunk 4]
```diff
@@ -268,7 +268,9 @@ func azureDevopsDisconnectCommand(p *Plugin, c *plugin.Context, commandArgs *mod
 }
 
 func executeDefault(p *Plugin, c *plugin.Context, commandArgs *model.CommandArgs, args ...string) (*model.CommandResponse, *model.AppError) {
-	return p.sendEphemeralPostForCommand(commandArgs, constants.InvalidCommand)
+	out := constants.InvalidCommand + "\n\n" + constants.HelpText
```
Comment (Kshitij-Katiyar) [inline server/plugin/command.go:271]: here you can add the "\n\n" with the invalid command constant as now you are not going to use this alone

File: server/serializers/subscriptions.go [hunk 5]
```diff
@@ -163,6 +163,20 @@ type Resource struct {
...
 	RefUpdates    []RefUpdates `json:"refUpdates"`
+	Fields        Fields       `json:"fields"`
+	Revision      Revision     `json:"revision"`
+}
+
+type Revision struct {
+	Fields Fields `json:"fields"`
+}
+
+type Fields struct {
```
Comment (Kshitij-Katiyar) [inline server/serializers/subscriptions.go:174]: If everything inside this struct is related to System its better to name this SystemFields I think

Comment (raghavaggarwal2308) [reply server/serializers/subscriptions.go:174]: I am using the same name returned from the API

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/api.go [hunk 6]
```diff
@@ -488,6 +489,8 @@ func (p *Plugin) getReviewersListString(reviewersList []serializers.Reviewer) st
 }
 
 func (p *Plugin) handleSubscriptionNotifications(w http.ResponseWriter, r *http.Request) {
+	// bdy, _ := ioutil.ReadAll(r.Body)
+	// fmt.Printf("\n\n\nbodyyy      %+v\n\n\n", string(bdy))
```
Comment (avas27JTG) [inline server/plugin/api.go:493]: remove these

**Review by ayusht2810 — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**